components/core/CMakeLists.txt

```diff
@@ -99,7 +99,7 @@ endif()
 # NOTE:
 # - We only try to link to the static library
 # - spdlog uses fmt, so their versions need to be kept in-sync
-find_package(fmt 8.0.1 REQUIRED)
+find_package(fmt 8.0.1 EXACT REQUIRED)
```

Comment:
It seems like the root cause is that fmt's CMake scripts don't work as expected---when we specify 8.0.1 here, fmt's CMake scripts should ensure that whatever version is found is compatible with the 8.0.1 APIs. However, that doesn't seem to be the case based on the error you saw. It also appears that other users have noted this [root cause](https://github.com/gabime/spdlog/pull/1890#issuecomment-896903372).

So instead, what we should probably do is find fmt, find spdlog, and then ensure they're compatible. Based on spdlog's release notes, these are compatible versions:

| spdlog | fmt  |
|--------|------|
| 1.15.x | 11.x |
| 1.12.x | 10.x |
| 1.11.x | 9.x  |
| 1.9.x  | 8.x  |